internal/worker/server.go

```diff
@@ -174,6 +174,94 @@ func (s *Server) CheckBuildDependencies(dep uuid.UUID, jobErr *clienterrors.Erro
 	return nil
 }
 
+// DependencyChainErrors recursively gathers all errors from job's dependencies,
```

Comment:
I always have mixed feelings about recursion but it's probably simpler than implementing DFS manually.

Reply:
I incline to keeping this as is. I think that recursion should be acceptable in this case, since we most probably won't have too deep job dependency chains  to make it an issue.